### PR TITLE
cap amount of work done at sync time

### PR DESCRIPTION
**Pull request by mmaxim**

### CODE REVIEW

File: go/chat/sync.go [hunk 1]
```diff
@@ -228,18 +228,34 @@ func (s *Syncer) handleFilteredConvs(ctx context.Context, uid gregor1.UID, syncC
...
 func (s *Syncer) getShouldUnboxSyncConvMap(ctx context.Context, convs []chat1.Conversation,
 	topicNameChanged []chat1.ConversationID) (m map[string]bool) {
 	m = make(map[string]bool)
 	for _, t := range topicNameChanged {
 		m[t.String()] = true
 	}
-	for _, conv := range convs {
-		if m[conv.GetConvID().String()] {
+	rconvs := utils.RemoteConvs(convs)
+	sort.Slice(rconvs, func(i, j int) bool {
```
Comment (joshblum) [inline go/chat/sync.go]: can you move this into utils so when we change the priority metric to unreadline (or something) we can change everything in one place?

File: go/chat/sync.go [hunk 2]
```diff
@@ -228,18 +228,34 @@ func (s *Syncer) handleFilteredConvs(ctx context.Context, uid gregor1.UID, syncC
...
+		return utils.GetConvMtime(rconvs[i]).After(utils.GetConvMtime(rconvs[j]))
+	})
+	maxConvs := s.maxSyncUnboxConvs()
+	for index, conv := range rconvs {
+		if index >= maxConvs {
```
Comment (joshblum) [inline go/chat/sync.go]:
```suggestion
		if len(m) >= maxConvs {
```

File: go/chat/sync.go (hunk 1 above)
Comment (mmaxim) [reply go/chat/sync.go]: I want this in time order, the point is to only unbox the stuff on the screen.

File: go/chat/sync.go [hunk 3]
```diff
@@ -228,18 +228,34 @@ func (s *Syncer) handleFilteredConvs(ctx context.Context, uid gregor1.UID, syncC
...
 		m[t.String()] = true
 	}
-	for _, conv := range convs {
-		if m[conv.GetConvID().String()] {
+	rconvs := utils.RemoteConvs(convs)
+	sort.Slice(rconvs, func(i, j int) bool {
+		return utils.GetConvMtime(rconvs[i]).After(utils.GetConvMtime(rconvs[j]))
+	})
+	maxConvs := s.maxSyncUnboxConvs()
+	for index, conv := range rconvs {
```
Comment (joshblum) [inline go/chat/sync.go]:
```suggestion
	for _, conv := range rconvs {
```

**Pull request merged**